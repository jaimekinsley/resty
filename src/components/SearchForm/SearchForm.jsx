import React from 'react';
import PropTypes from 'prop-types';

const SearchForm = ({ api, jsonBody, method, onChange, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <input type="text" name="api" placeholder="API URL" value={api} onChange={onChange} />

    <label>
      <input type="radio" name="method" value="get" checked={method === 'get'} onChange={onChange}/>
        GET
    </label>
    <label>
      <input type="radio" name="method" value="post" checked={method === 'post'} onChange={onChange} />
        POST
    </label>
    <label>
      <input type="radio" name="method" value="put" checked={method === 'put'} onChange={onChange}/>
        PUT
    </label>
    <label>
      <input type="radio" name="method" value="patch" checked={method === 'patch'} onChange={onChange} />
        PATCH
    </label>
    <label>
      <input type="radio" name="method" value="delete" checked={method === 'delete'} onChange={onChange}/>
        DELETE
    </label>

    <input type="textarea" name="jsonBody" value={jsonBody} onChange={onChange} />

    <button>Submit</button>
  </form>
);

SearchForm.propTypes = {
  api: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  jsonBody: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default SearchForm;
