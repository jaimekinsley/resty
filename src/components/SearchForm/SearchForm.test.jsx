import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm component', () => {
  let handleChange;
  let handleSubmit;
  let wrapper;

  beforeEach(() => {
    handleChange = jest.fn();
    handleSubmit = jest.fn();
    wrapper = shallow(<SearchForm
      api="https://www.futuramaapi.herokuapp.com"
      method="patch"
      jsonBody="{ 'name': 'bender' }"
      onChange={handleChange}
      onSubmit={handleSubmit}
    />);
  });

  it('has an input with a value set to api', () => {
    expect(wrapper.find('input[name="api"]').prop('value')).toEqual('https://www.futuramaapi.herokuapp.com');
  });

  it('has an radio button with a value set to PATCH', () => {
    expect(wrapper.find('input[name="method"]').findWhere(radios => radios.prop('checked')).prop('value')).toEqual('patch');
  });

});

