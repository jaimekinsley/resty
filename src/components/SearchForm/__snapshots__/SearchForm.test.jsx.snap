// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders the SearchForm component 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <input
    name="api"
    onChange={[MockFunction]}
    placeholder="API URL"
    type="text"
    value="https://www.futuramaapi.herokuapp.com"
  />
  <label>
    <input
      checked={false}
      name="method"
      onChange={[MockFunction]}
      type="radio"
      value="get"
    />
    GET
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[MockFunction]}
      type="radio"
      value="post"
    />
    POST
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[MockFunction]}
      type="radio"
      value="put"
    />
    PUT
  </label>
  <label>
    <input
      checked={true}
      name="method"
      onChange={[MockFunction]}
      type="radio"
      value="patch"
    />
    PATCH
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[MockFunction]}
      type="radio"
      value="delete"
    />
    DELETE
  </label>
  <input
    name="jsonBody"
    onChange={[MockFunction]}
    type="textarea"
    value="{}"
  />
  <button>
    Submit
  </button>
</form>
`;
